operator/src/main/java/oracle/kubernetes/weblogic/domain/model/Domain.java

```diff
@@ -501,10 +502,144 @@ public boolean isHttpAccessLogInLogHome() {
     return spec.getHttpAccessLogInLogHome();
   }
 
+  /**
+   * Returns if the domain is using online update.
+   * return true if using onlne update
+   */
+
+  public boolean isUseOnlineUpdate() {
+    return Optional.ofNullable(spec)
+        .map(DomainSpec::getConfiguration)
+        .map(Configuration::getModel)
+        .map(Model::getOnlineUpdate)
+        .map(OnlineUpdate::getEnabled)
+        .orElse(false);
+  }
+
+  /**
+   * Returns WDT activate timeout.
+   * @return WDT activate timeout
+   */
+  public Long getWDTActivateTimeout() {
+    return Optional.ofNullable(spec)
+        .map(DomainSpec::getConfiguration)
+        .map(Configuration::getModel)
+        .map(Model::getOnlineUpdate)
+        .map(OnlineUpdate::getActivateTimeoutMilliSeconds)
+        .orElse(null);
```

Comment:
Minor code refactor:  There are several methods here with repetitious code where the only difference is the method reference that's passed into the last 'map'.  Is it possible to create a helper method getWDTTimeout(online-update-method-ref) so that this method implementation becomes getWDTActivateTimeout(OnlineUpdate::getActivateTimeoutMilliSeconds)...    Or is it possible to remove these getters, and only have the helper, and have anyh callers call the helper directly?